services/workflows-service/src/customer/customer.service.ts

```diff
@@ -48,10 +58,35 @@ export class CustomerService {
   }
 
   async updateById(id: string, args: Parameters<CustomerRepository['updateById']>[1]) {
-    return (await this.repository.updateById(id, args)) as unknown as TCustomerWithFeatures;
+    return await this.prisma.$transaction(async transaction => {
+      const customer = (await this.repository.updateById(
+        id,
+        args,
+        transaction,
+      )) as unknown as TCustomerWithFeatures;
+
+      await retry(() => new UnifiedApiClient().updateCustomer(id, customer));
+
+      return customer;
+    });
   }
 
   async deleteById(id: string, args?: Parameters<CustomerRepository['deleteById']>[1]) {
-    return this.repository.deleteById(id, args);
+    return await this.prisma.$transaction(async transaction => {
+      await this.repository.deleteById(id, args, transaction);
+
+      await retry(() => new UnifiedApiClient().deleteCustomer(id));
+    });
   }
 }
+
+const retry = async (fn: () => Promise<unknown>) => {
+  const { default: pRetry } = await import('p-retry');
```

Comment:
This package supports ESM only, and we use CJS, so I used dynamic import to overcome this issue.